src/utils.rs

```diff
@@ -24,6 +24,28 @@ pub fn format_duration(d: &std::time::Duration) -> String {
     s.trim_end().to_string()
 }
 
+/// Returns a human readable String of milliseconds in the HH:MM:SS format.
+pub fn ms_to_hms(duration: u32) -> String {
+    let mut formated_time = String::new();
+
+    let total_seconds = duration / 1000;
+    let seconds = total_seconds % 60;
+    let minutes = (total_seconds / 60) % 60;
+    let hours = total_seconds / 3600;
+
+    if hours > 0 {
+        formated_time.push_str(&format!("{hours}:"));
+    }
+    if hours > 0 {
+        formated_time.push_str(&format!("{minutes:02}:"));
```

Comment:
Could this block be merged?

Reply:
Oh oops. I was thinking in terms of the steps that had to be applied (hours, minutes, seconds) to get to the representation. Didn't consider moving those together. :smile: